src/javascripts/base_app.js

```diff
@@ -95,6 +97,8 @@ BaseApp.prototype = {
 
   switchTo: function(name, data) {
     this.$('[data-main]').html(this.renderTemplate(name, data));
+    var newHeight = Math.min($('html').height(), maxHeight);
+    this.zafClient.invoke('resize', { height: newHeight, width: '100%' });
```

Comment:
we might also want to patch this behaviour into jquery functions that manipulate the dom...